### PR TITLE
Made OptionsProcessors classes static

### PR DESCRIPTION
**Pull request by JudeVajira**
Made OptionsProcessors classes static as it would remove the need to create an object when it is needed.

### CODE REVIEW

File: AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs [hunk 1]
```diff
@@ -27,7 +27,6 @@ private static IEnumerable<SearchTerm> GetAllTerms(IEnumerable<DTColumn> columns
                 }
 
                 var hasNavigation = column.Data.Contains('.');
-                var parentIndex = column.Data.Split('.').Length - 2;
```
Comment (fingers10) [inline AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs:30]: I still wonder how my vs 2019 failed to highlight unused variable.

File: AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs [hunk 2]
```diff
@@ -51,12 +50,11 @@ private static IEnumerable<SearchTerm> GetValidTerms(IEnumerable<DTColumn> colum
                 yield break;
             }
 
-            var declaredTerms = GetTermsFromModel(typeof(T));
+            var declaredTerms = GetTermsFromModel(typeof(T)).ToList();
```
Comment (fingers10) [inline AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs:53]: Please remove `ToList()` we can add that if we are enumerating this more than once.

File: AspNetCoreWeb/Infrastructure/SortOptionsProcessor{T,TEntity}.cs [hunk 3]
```diff
@@ -46,12 +45,11 @@ private static IEnumerable<SortTerm> GetValidTerms(JqueryDataTablesParameters ta
                 yield break;
             }
 
-            var declaredTerms = GetTermsFromModel(typeof(T));
+            var declaredTerms = GetTermsFromModel(typeof(T)).ToList();
```
Comment (fingers10) [inline AspNetCoreWeb/Infrastructure/SortOptionsProcessor{T,TEntity}.cs:48]: Please remove `ToList()` we can add that if we are enumerating this more than once.

**Review by fingers10 — COMMENTED** [review]
Thanks for the PR. Much appreciated. I'll try to externalize excel export to target .net standard 2.0.

File: AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs (hunk 2 above)
Comment (JudeVajira) [reply AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs:53]: Calling SingleOrDefault on line 57 does iterate the entire list right? Or am I mistaken?

File: AspNetCoreWeb/Infrastructure/SortOptionsProcessor{T,TEntity}.cs (hunk 3 above)
Comment (JudeVajira) [reply AspNetCoreWeb/Infrastructure/SortOptionsProcessor{T,TEntity}.cs:48]: Calling SingleOrDefault on line 52 does iterate the entire list right? Or am I mistaken?

File: AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs (hunk 2 above)
Comment (fingers10) [reply AspNetCoreWeb/Infrastructure/SearchOptionsProcessor{T,TEntity}.cs:53]: I'm sorry. You're right. I misunderstood.

File: AspNetCoreWeb/Infrastructure/SortOptionsProcessor{T,TEntity}.cs (hunk 3 above)
Comment (fingers10) [reply AspNetCoreWeb/Infrastructure/SortOptionsProcessor{T,TEntity}.cs:48]: I'm sorry. You're right. I misunderstood.

**Pull request merged**